proto/dataplane.proto

```diff
@@ -0,0 +1,211 @@
+syntax = "proto3";
+
+package config;
+
+option go_package = "dataplane/grpc";
+
+service ConfigService {
+  rpc GetConfig (GetConfigRequest) returns (GatewayConfig);
+  rpc GetConfigGeneration (GetConfigGenerationRequest) returns (GetConfigGenerationResponse);
+  rpc UpdateConfig (UpdateConfigRequest) returns (UpdateConfigResponse);
+}
+
+message GetConfigRequest {
+}
+
+message UpdateConfigRequest {
+  GatewayConfig config = 1;
+}
+
+message UpdateConfigResponse {
+  Error error = 1;
+  string message = 2;
+}
+
+enum Error {
+  ERROR_NONE = 0;
+  ERROR_VALIDATION_FAILED = 1;
+  ERROR_APPLY_FAILED = 2;
+  ERROR_UNKNOWN_ERROR = 3;
+}
+
+message GetConfigGenerationRequest {
+}
+
+message GetConfigGenerationResponse {
+  uint64 generation = 1;
+}
+
+/* ================================================ */
+/* ==== Interfaces (both underlat and overlay) ==== */
+/* ================================================ */
+
+message Interface {
+  string name = 1;
+  string ipaddr = 2;
+  IfType type = 3;
+  IfRole role = 4;
+}
+
+enum IfType {
+  IF_TYPE_ETHERNET = 0;
+  IF_TYPE_VLAN = 1;
+  IF_TYPE_LOOPBACK = 2;
+  IF_TYPE_VTEP = 3;
+}
+
+enum IfRole {
+  IF_ROLE_FABRIC = 0;
+  IF_ROLE_MANAGEMENT = 1;
```

Comment:
I don't think management iface is needed for dataplane. What's the use case for it?